src/middlewares/searchDispatcher.js

```diff
@@ -0,0 +1,30 @@
+import { LOCATION_CHANGE } from 'react-router-redux';
+import { searchForCurrentLocation } from '../actions/search';
+
+function getLocationFromRootState(state) {
+  const { router } = state;
+  const { location } = router;
```

Comment:
why not 

```js
const { router: {location}  } = state;
```

Reply:
just personal preference :), but I'll do that